test/Libraries/DynamoMSOfficeTests/ExcelTests.cs

```diff
@@ -935,6 +1182,351 @@ public void TestWriteFunctionObjectToExcel()
             // Empty list expected
             Assert.AreEqual(0, list.Count);
         }
+        [Test]
+        public void WriteToNonExistingFile()
+        {
+
+            // 1.  Write data into non existing file  and Run
+            // 2.  Dynamo must create the file and write the data
+
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteNonExistingFile.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            var watch = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("8651eeba-33a9-4704-b92c-346e6e5e4b34");
+            ViewModel.HomeSpace.Run();
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+            var list2 = watch.CachedValue.GetElements();
+            Assert.AreEqual(5, list2.Count());
+            //   1, 2, 3, 4,5
+            Assert.IsTrue(list2[0].IsCollection);
+            var rowList1 = list2[0].GetElements();
+            Assert.AreEqual(1, rowList1[0].Data);
+            var rowList2 = list2[1].GetElements();
+            Assert.AreEqual(2, rowList2[0].Data);
+            var rowList3 = list2[2].GetElements();
+            Assert.AreEqual(3, rowList3[0].Data);
+            var rowList4 = list2[3].GetElements();
+            Assert.AreEqual(4, rowList4[0].Data);
+            var rowList5 = list2[4].GetElements();
+            Assert.AreEqual(5, rowList5[0].Data);
+        }
+        [Test]
+        public void WriteNonExistingSheet()
+        {
+
+            // 1.  Open Sample ImportExport_Excel_ToDynamo.dyn and Run
+            // 2.  NonExisting Sheet
+            //3.  Rerun the file  
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteNonExistingSheet.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            var watch = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("32dbe300-4780-4c47-b07c-63019d5285ac");
+            ViewModel.HomeSpace.Run();
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+            var list2 = watch.CachedValue.GetElements();
+            Assert.AreEqual(5, list2.Count());
```

Comment:
Just use `Count` with `List`s. This should be more efficient unless `Count()` is overridden for `List`s